src/hypergol/tensorflow_model_manager.py

```diff
@@ -57,62 +57,69 @@ def restore_model_weights(self):
         """ restores tensorflow model weights """
         self.model.load_weights(f'{self.restoreWeightsPath}/{self.model.get_name()}.h5')
 
-    def initialize(self):
-        """
-        initialization method, restores variables and creates summary writers
-        """
-        self.trainingSummaryWriter = tf.summary.create_file_writer(logdir=f'{self.tensorboardPath}/train')
-        self.evaluationSummaryWriter = tf.summary.create_file_writer(logdir=f'{self.tensorboardPath}/evaluate')
-        if self.restoreWeightsPath is not None:
-            self.evaluate(withLogging=False, withMetadata=False)  # model call needed to initialize layers/weights before reloading
-            self.model.restore_model_weights(path=self.restoreWeightsPath)
-
-    def train(self, withLogging, withMetadata):
+    def train(self, withTracing):
         """runs a training step for the model
 
         Parameters
         ----------
-        withLogging: bool
-            produce tensorflow logging for step
         withMetadata: bool
             log tensorflow graph metadata for step
         """
-        batch = next(self.batchProcessor)
-        if withMetadata and self.globalStep > 0:
+        inputs, targets = next(self.batchProcessor)
+        if withTracing:
             tf.summary.trace_on(graph=True, profiler=False)
-        loss = self.model.train(inputs=batch['inputs'], targets=batch['targets'], optimizer=self.optimizer)
-        if withLogging:
-            with self.trainingSummaryWriter.as_default():
-                tf.summary.scalar(name='Loss', data=loss, step=self.globalStep)
-                if withMetadata and self.globalStep > 0:
-                    tf.summary.trace_export(name=f'{self.model.get_name()}{self.globalStep}', step=self.globalStep, profiler_outdir=f'{self.tensorboardPath}/trainGraph')
+
+        with tf.GradientTape() as tape:
+            loss = self.model.get_loss(targets=targets, training=True, **inputs)
+        grads = tape.gradient(loss, self.model.trainable_variables)
+        self.optimizer.apply_gradients(zip(grads, self.model.trainable_variables))
+
+        # Watch this issue: https://github.com/PyCQA/pylint/issues/3596
+        with self.trainingSummaryWriter.as_default():                   #pylint: disable=not-context-manager
+            tf.summary.scalar(name='Loss', data=loss, step=self.globalStep)
+            if withTracing:
+                tf.summary.trace_export(
+                    name=f'{self.model.get_name()}{self.globalStep}',
+                    step=self.globalStep,
+                    profiler_outdir=f'{self.tensorboardPath}/trainGraph'
+                )
         self.globalStep += 1
         return loss
 
-    def evaluate(self, withLogging, withMetadata):
+    def evaluate(self, withTracing):
         """runs an evaluation step for the model
 
         Parameters
         ----------
-        withLogging: bool
-            produce tensorflow logging for step
         withMetadata: bool
```

Comment:
param needs renaming here too

Reply:
done but comments needs a general scrubbing at the end